### ISSUE TITLE
Unexpected modal dialogue, in lieu of banner 'Master password prompt was blocked'

### ISSUE THREAD

**Issue by grahamperrin** [vanowm/MasterPasswordPlus#146]
For example: 

![2018-09-29 09 37 23 sign in continue to firefox add-ons - waterfox](https://user-images.githubusercontent.com/192271/46243425-279ac800-c3cc-11e8-9341-df79502fe2d3.png)

With a preference to block _all_ master password prompts

It seems that the banner will not appear unless preceded (during a session – after a start of the browser) by, for example, these steps: 

1. click the address bar icon for Master Password+ 
2. if the modal dialogue does not appear then again, click the icon
3. escape from the dialogue.

<strike>Please, is this expected behaviour? </strike>

## Environment

- Waterfox 56.2.3
- FreeBSD-CURRENT
- Master Password+ 1.30.6 with a handful of other extensions

**Comment by grahamperrin** [comment]
> … with a handful of other extensions …

[Extension Conflict Troubleshooter](http://softwarebychuck.com/fect/fect.html) determined that there's no conflict. 

----

> … It seems that the banner will not appear unless preceded … by, for example, these steps: …

There's an alternative workaround: 

1. open a new window
2. **refrain from using the initial window**

– tested with a logged-out view of https://addons.mozilla.org/user/vanowm/

Via <https://www.reddit.com/comments/8l2yrz/-/dzddqt0/>: 

- Mozilla bug [1378459 - (webextensions-startup) Allow some addon functionality to load prior to any content loading](https://bugzilla.mozilla.org/show_bug.cgi?id=1378459)

From <https://www.reddit.com/r/waterfox/comments/8l2yrz/-/dziex4f/>: 

> Hm, odd that a private window has to be launched. … 

@MrAlex94 I think _private_ was a red herring. See workaround step (1) in this comment; a simple new window suffices.

**Comment by vanowm** [comment]
I'm sorry, it must be too early for me, but I don't understand at all what this issue is about...
Clicking MP+ buttons will never block any prompts, only prompts that were initialized by other parts of browser/extensions supposed to be blocked.

**Comment by grahamperrin** [comment]
> too early for me, 

:-)

This screen recording _might_ help to visualise the issue. Note the situations in which the blue banner appears (or does not appear) when the Home icon is clicked: 

[2018-09-30 10:29.zip](https://github.com/vanowm/MasterPasswordPlus/files/2431455/2018-09-30.10.29.zip)

– _home_ for that recording was `https://bugzilla.mozilla.org`

**Comment by vanowm** [comment]
Thank you, this helped a lot to understand the problem.

Please turn on debug in MP+ options (MP+ Options -> Help -> Debug level -> debug) and provide messages from browser console (CTRL+SHIFT+J) after browser was restarted and MP prompt showed.

**Comment by grahamperrin** [comment]
Thanks, there's yesterday's question alongside the (2017) e10s-related code comment at https://github.com/vanowm/MasterPasswordPlus/commit/610d630520a4a1aa7da170e4808849f6a0664ec6#diff-3d89c5bc11f5638315723c6760cd6d40R1833

Already tested a few times with Waterfox in single-process mode. The screen recording above was of the app in that mode. 

> … can't reproduce this issue on Windows system

Interesting. Hmm. Have you tested with a new profile and the extension alone, restricting yourself to a single window (not opening any new window) after the app is started?

**Comment by grahamperrin** [comment]
> … Please turn on debug in MP+ options … Debug level -> debug…

Ah, doing so makes the issue non-reproducible :-)

Can you reproduce the issue on Windows with debugging _disabled_?

**Comment by vanowm** [comment]
Still can't reproduce it on Windows.
Is there any messages in browser console ?

The only thing debug does different is open browser console, otherwise it doesn't change anything else...so from the looks of it, MP+ doesn't initialize properly on startup.

**Comment by vanowm** [comment]
Try v1.30.7b1

**Comment by grahamperrin** [comment]
Thanks! 

That fixes it for me. Tested with multi-process enabled (although re: https://github.com/vanowm/MasterPasswordPlus/commit/610d630520a4a1aa7da170e4808849f6a0664ec6#diff-3d89c5bc11f5638315723c6760cd6d40R1833 I don't know whether multi-process is acceptable for all other features of the extension). 

Do you know, was Mozilla bug 1378459 a factor, or was my guess way off?

**Comment by vanowm** [comment]
That bug is webextensions related, MP+ is "legacy" xul extension, it runs way before even bootstrap extension does, so no I don't believe that bug is related.

As of multiprocess, it was an attempt add ability trigger MP prompt when specific address (chrome or web) was opened.

**Comment by grahamperrin** [comment]
> That bug is webextensions related, …

Yeah … the classification didn't occur until after Firefox Quantum was released. 

I was fairly sure that I found a _general_ symptom – **initial window not working reliably for some types of extension** (or words to that effect) – affecting legacy extensions as well as non-legacy. Today I can't reproduce that exactly with the legacy example that I gave in the bug ([Tab Scope](https://web.archive.org/web/20181003052101/https://addons.mozilla.org/en-GB/firefox/addon/tab-scope/)). In retrospect I _could_ have kept a better record of other affected extensions, but TBH I just got myself in the habit, well over a year ago, of studiously avoiding the initial window :-) to avoid the bug. 

Thanks again. Nice pigtails.
